src/Tests/dotnet.Tests/dotnet-workload-clean/GivenDotnetWorkloadClean.cs

```diff
@@ -0,0 +1,198 @@
+{
+    public class GivenDotnetWorkloadClean : SdkTest
+    {
+        private readonly BufferedReporter _reporter;
+        private readonly string _manifestPath;
+
+        private MockWorkloadManifestUpdater _manifestUpdater = new();
+        private readonly string _sdkFeatureVersion = "6.0.100";
+        private readonly string _installingWorkload = "xamarin-android";
+        private readonly string dotnet = nameof(dotnet);
+        private readonly string _profileDirectoryLeafName = "user-profile";
+
+        public GivenDotnetWorkloadClean(ITestOutputHelper log) : base(log)
+        {
+            _reporter = new BufferedReporter();
+            _manifestPath = Path.Combine(_testAssetsManager.GetAndValidateTestProjectDirectory("SampleManifest"), "Sample.json");
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(false, false)]
+        public void GivenWorkloadCleanFileBasedItRemovesPacksAndPackRecords(bool userLocal, bool cleanAll)
+        {
+            // Setup ( I wish we could not copy this everywhere, but these are convenient as locals. )
+            var testDirectory = _testAssetsManager.CreateTestDirectory(identifier: userLocal ? $"userlocal-{cleanAll}" : $"default-{cleanAll}").Path;
+            var dotnetRoot = Path.Combine(testDirectory, dotnet);
+            var userProfileDir = Path.Combine(testDirectory, _profileDirectoryLeafName);
+            var workloadResolver = WorkloadResolver.CreateForTests(new MockManifestProvider(new[] { _manifestPath }), dotnetRoot, userLocal, userProfileDir);
+            var nugetDownloader = new MockNuGetPackageDownloader(dotnetRoot);
+
+            string installRoot = userLocal ? userProfileDir : dotnetRoot;
+            if (userLocal)
+            {
+                WorkloadFileBasedInstall.SetUserLocal(dotnetRoot, _sdkFeatureVersion);
+            }
+
+            // Test
+            InstallWorkload(userProfileDir, dotnetRoot, testDirectory, workloadResolver, nugetDownloader);
+
+            var extraPackRecordPath = MakePackRecord(installRoot);
+            var extraPackPath = MakePack(installRoot);
+
+
+            var cleanCommand = cleanAll ? GenerateWorkloadCleanAllCommand(workloadResolver, userProfileDir, dotnetRoot) : GenerateWorkloadCleanCommand(workloadResolver, userProfileDir, dotnetRoot);
+            cleanCommand.Execute();
+
+            AssertExtraneousPacksAreRemoved(extraPackPath, extraPackRecordPath);
+            AssertValidPackCountsMatchExpected(installRoot, expectedPackCount: cleanAll ? 0 : 7, expectedPackRecordCount: cleanAll ? 0 : 8);
+
+            AssertAdjacentCommandsStillPass(userProfileDir, dotnetRoot, testDirectory, workloadResolver, nugetDownloader);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GivenWorkloadCleanAllFileBasedItCleansOnlyOlderAndCurrentFeatureBands(bool userLocal)
```

Comment:
Is this test name correct?  I think we changed it so that clean all would remove workloads from future bands, and the last line of the test verifies that no packs are installed.  So probably the name should be updated to something like `GivenWorkloadCleanAllFileBasedItCleansOlderAndNewerFeatureBands`